src/classes/BDI_ContactService.cls

```diff
@@ -317,7 +317,9 @@ public with sharing class BDI_ContactService {
             List<String> listDiKey = listDIKeyCx(di, strCx);
             Contact con = ContactFromDiKeys(listDiKey);
             if (con != null) {
-                if (con.Id != null && con.AccountId != null && con.Account.npe01__SYSTEM_AccountType__c != CAO_Constants.HH_ACCOUNT_TYPE) {
+                Boolean contactAccountIsNotHousehold = con.Id != null && con.AccountId != null 
```

Comment:
I would consider expanding this to include administrative accounts. If you had contacts associated with an organization in ADV you would still want to throw an error.

Reply:
I could be wrong, but I feel like in HEDA you could potentially have a Donor Contact with an Academic Program as their Account (Professor donates back to the school?). @kevinswiggum might know if this is a potential use case?

Reply:
I believe BDI would handle that scenario incorrectly - see comments in NPSP Data Import issues when using the Admin Account model in HEDA + Advancement. You should include tests for expected behavior for advancement in addition to bypassing the NGO tests since you're supporting new functionality.

Reply:
@lparrott  Regarding the faculty member scenario...we'll probably need to be a bit more prescriptive in how they link contacts to accounts when Advancement is installed. Otherwise you'll start having personal donations rolling up to Department accounts. I think it's fair to steer customers in a direction where faculty either have Administrative or Household accounts and associate them to their Department account with Affiliations.

Reply:
@kevinswiggum Great feedback, thanks for the quick response! We discussed this in Seattle this week and agree that only Contacts with Household or Administrative Accounts should be able to give donations.